build-system/tasks/helpers.js

```diff
@@ -457,6 +443,69 @@ function compileJs(srcDir, srcFilename, destDir, options) {
   }
 }
 
+/**
+ * Enables local testing mode for various target files
+ * @param {string} destFilename
+ */
+function maybeEnableLocalTesting(destFilename) {
+  if (process.env.NODE_ENV === 'development') {
+    if (destFilename === 'amp.js') {
+      return enableLocalTesting('dist/amp.js');
+    } else if (destFilename === 'amp-esm.js') {
+      return enableLocalTesting('dist/amp-esm.js');
+    } else if (destFilename === 'amp4ads-v0.js') {
+      return enableLocalTesting('dist/amp4ads-v0.js');
+    } else if (destFilename === 'integration.js') {
+      return enableLocalTesting('dist.3p/current/integration.js');
+    } else if (destFilename === 'amp-shadow.js') {
+      return enableLocalTesting('dist/amp-shadow.js');
+    } else if (destFilename === 'amp-inabox.js') {
+      return enableLocalTesting('dist/amp-inabox.js');
+    } else {
+      return Promise.resolve();
+    }
+  } else {
+    return Promise.resolve();
+  }
+}
+
+/**
+ * Transpiles from TypeScript into intermediary files before compilation and
+ * deletes them afterwards.
+ *
+ * @param {string} srcDir Path to the src directory
+ * @param {string} srcFilename Name of the JS source file
+ * @param {string} destDir Destination folder for output script
+ * @param {?Object} options
+ * @return {!Promise}
+ */
+function compileTs(srcDir, srcFilename, destDir, options) {
```

Comment:
This function is new and you're exporting it, but the PR doesn't modify any other file so it remains unused. The base file also has 4 uses of `options.typescript`, which the new file does not. Which one of us missed something here? 😝

Reply:
Good observation, fair comment.

@choumx added code to compile TypeScript extensions in #13792, which used `options.typescript` as the entry point. This is not in use by any extension today.

In the interest of streamlining `compileJs`, I separated all TS functionality into `compileTs`. This is not in use either, but is available when the first TS extension is written. With this, `compileJs` needn't concern itself with TS extensions, and the intention at the calling site is made clearer.

Reply:
It might be prudent to summarize this reply in the code for future reference

Reply:
The new code has JSDoc at the top. The previous version was not in use. I've added @choumx to review this part.